tests/unit_tests/DescriptorSetAdd_test.cc

```diff
@@ -218,6 +218,119 @@ TEST(Descriptors_Add, add_flatl2_100d_2add) {
   delete[] xb;
 }
 
+//HNSW Tests 
```

Comment:
A few comments on what is going on overall in each test, & comments in a couple of the steps, would help a reader follow along. especially since folks would likely look at the test cases to figure out how to use the API

Reply:
added few comments. The unit tests for hnsw follows the same pattern of the defined tests for all the other indices used in VDMS.